timesketch/api/v1/resources/timeline.py

```diff
@@ -198,10 +230,61 @@ def post(self, sketch_id, timeline_id):
                 HTTP_STATUS_CODE_FORBIDDEN,
                 'The user does not have write permission on the sketch.')
 
+        form = forms.TimelineForm.build(request)
         if not form.validate_on_submit():
             abort(
                 HTTP_STATUS_CODE_BAD_REQUEST, 'Unable to validate form data.')
 
+        if form.labels.data:
+            label_string = form.labels.data
+            labels = json.loads(label_string)
+            if not isinstance(labels, (list, tuple)):
+                abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST, (
+                        'Label needs to be a JSON string that '
+                        'converts to a list of strings.'))
+            if not all([isinstance(x, str) for x in labels]):
+                abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST, (
+                        'Label needs to be a JSON string that '
+                        'converts to a list of strings (not all strings)'))
+
+            label_action = form.label_action.data
+            if label_action not in ('add', 'remove'):
```

Comment:
Why don't we use proper HTTP verbs here? e.g.

http.delet()
http.post()

I know this is how we have designed the backend API, but I'm thinking if we would like to clean that up later. Not in this PR obviously :)

Reply:
ack